### PR TITLE
Add a deprecation warning when building with C++17

### PR DESCRIPTION
**Pull request by jmcarcell**
BEGINRELEASENOTES
- Add a deprecation warning when building with C++17 before removing support for it in https://github.com/AIDASoft/podio/pull/698

ENDRELEASENOTES

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -40,6 +40,10 @@ if(NOT CMAKE_CXX_STANDARD MATCHES "17|20")
   message(FATAL_ERROR "Unsupported C++ standard: ${CMAKE_CXX_STANDARD}")
 endif()
 
+if(CMAKE_CXX_STANDARD EQUAL 17)
+  message(WARNING "C++17 is deprecated and support for it will be removed in the future. Please use at least C++20.")
```
Comment (tmadlener) [inline CMakeLists.txt:44]: Should we put a version here when we are going to remove it? If we want to have a tag in between where this message should be visible that would make it v01-03.

Comment (jmcarcell) [reply CMakeLists.txt]: Added as discussed today in the podio and EDM4hep meeting.

**Comment by andresailer** [comment]
Should we also set the default standard to 20 then, and not print the warning if the user doesn't do anything?

(cannot comment on the line directly)
https://github.com/AIDASoft/podio/pull/700/files#diff-1e7de1ae2d059d21e1dd75d5812d5a34b0222cef273b7c3a2af62eb747f9d20aR37

**Comment by tmadlener** [comment]
> Should we also set the default standard to 20 then, and not print the warning if the user doesn't do anything?

Yes, definitely.

**Comment by jmcarcell** [comment]
Added, I thought leaving all the changes to #698.

**Comment by tmadlener** [comment]
The `dev3` based workflows are failing for several PRs and I think is unrelated to the changes in any of those. From what I can tell from just looking at the outputs, it looks like we no longer recognize whether a file has a TTree or an RNTuple inside and it then instantiates the wrong reader in the interface:

https://github.com/AIDASoft/podio/actions/runs/11681697887/job/32527272599?pr=700#step:4:855

**Pull request merged**